### PR TITLE
GAWB-3216 add logging for notebook calls

### PR DESCRIPTION
**Pull request by ahaessly**
diff from ah-leo-local for easier compare

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by ahaessly** [comment]
here is the additional matcher needed by logstash. please comment on this too if you have feedback.
"%{NOTSPACE:skip1} %{NOTSPACE:skip2} %{NOTSPACE:skip3} %{NOTSPACE:skip4} %{NOTSPACE:skip5} %{IPORHOST:ip} %{HTTPDUSER:ident} %{HTTPDUSER:auth} \[%{TIMESTAMP_ISO8601:timestamp}\] \"(?:%{WORD:verb} %{NOTSPACE:request}(?: HTTP/%{NUMBER:httpversion})?|%{DATA:rawrequest})\" %{NUMBER:response} (?:%{NUMBER:bytes:int}|-) (?:%{NOTSPACE:server}|-) %{GREEDYDATA:browser_name}"

**Review by davidangb — COMMENTED** [review]
if you're going the route of writing your own logstash grok pattern for these logs, :+1:.

if you're relying on an existing logstash grok pattern, then the log you're generating seems to have some fields out of order/missing - see https://github.com/broadinstitute/firecloud-develop/blob/dev/base-configs/firecloud-orchestration/site.conf#L7.

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutes.scala [hunk 1]
```diff
@@ -103,6 +108,28 @@ trait ProxyRoutes extends UserInfoDirectives with CorsSupport { self: LazyLoggin
...
+  private def logRequestResultForMetrics(userInfo: UserInfo): Directive0 = {
+    def myMetricsLogger(logger: LoggingAdapter)(req: HttpRequest)(res: Any): Unit = {
+      val headers = req.headers
+      val headerMap: Map[String, String] = headers.map { header =>
+        (header.name(), header.value())
+      }.toMap
+
+      val entry = res match {
+        case Complete(resp) =>
+          LogEntry(s"${headerMap.getOrElse("X-Forwarded-For", "0.0.0.0")} ${userInfo.userEmail} " +
```
Comment (rtitle) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutes.scala]: I am wondering if this matches exactly with https://github.com/broadinstitute/firecloud-develop/blob/dev/base-configs/firecloud-orchestration/site.conf#L6-L7 (does it need to?).

I ran `ProxyRoutesSpec` against this branch and saw the following logs:
```
[INFO] [03/15/2018 13:58:58.201] [org-broadinstitute-dsde-workbench-leonardo-api-ProxyRoutesSpec-akka.actor.default-dispatcher-19] [akka.actor.ActorSystemImpl(org-broadinstitute-dsde-workbench-leonardo-api-ProxyRoutesSpec)] 0.0.0.0 user1@example.com 1234567890 [2018-03-15T17:58:58] "POST http://example.com/notebooks/dsp-leo-test/test HTTP/1.1" 200 177 - unknown
```
whereas the ones logged by Apache look like:
```
Mar 14 11:07:41 gce-leonardo-dev101 docker-current/leonardo-proxy[6204]: 69.173.127.109 - - [14/Mar/2018:11:07:41 -0400] "PUT /notebooks/broad-dsde-dev/anu-notebooks-name-bug/api/contents/anu-nb.ipynb HTTP/1.1" 200 227 "https://leonardo.dsde-dev.broadinstitute.org/notebooks/broad-dsde-dev/anu-notebooks-name-bug/notebooks/anu-nb.ipynb" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.162 Safari/537.36"
```

One difference is that this prepends the patten specified in logback.xml: https://github.com/DataBiosphere/leonardo/blob/develop/src/main/resources/logback.xml#L23

Would it be better to create a new logback appender for these logs which don't use this pattern?

(Or if it's OK to not match the Apache format exactly then we could leave it as is.)

**Review by rtitle — APPROVED** [review]
Discussed in person. We can keep this format (with a new grok pattern). Also asked if we can have a unit test for this :) Would require intercepting log messages with a mock or something. Not completely sure how feasible that is.

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
jenkins retest

**Review by rtitle — APPROVED** [review]
I went ahead and rebased this to develop. Will merge once tests pass.

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
jenkins retest

**Pull request merged**